### PR TITLE
post: go-ipfs 0.4.22

### PR DESCRIPTION
**Pull request by Stebalien**

### CODE REVIEW

File: content/post/054-go-ipfs-0.4.22.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+---
+date: 2019-08-06
+url: 93-go-ipfs-0.4.22
+title: go-ipfs 0.4.22 released
+author: Steven Allen
+---
+
+We're releasing a PATCH release of go-ipfs based on 0.4.21 containing some critical fixes.
+
```
Comment (Stebalien) [inline content/post/054-go-ipfs-0.4.22.md]: Let's release this post after the release process post so we can link to https://github.com/ipfs/blog/pull/266.

File: content/post/054-go-ipfs-0.4.22.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+---
+date: 2019-08-06
+url: 054-go-ipfs-0.4.22
+title: go-ipfs 0.4.22 released
+author: Steven Allen
+---
+
+We're releasing a PATCH release of go-ipfs based on 0.4.21 containing some critical fixes.
+
+The past several releases have been shaky and the network has scaled to the point where where small changes can have a wide-reaching impact on the entire network. To keep this situation from escalating, we've put a hold on releasing new features until we can improve our [release process](https://github.com/ipfs/go-ipfs/pull/6482) (which we will be trialing in this release) and [testing procedures](https://github.com/ipfs/go-ipfs/issues/6483).
```
Comment (momack2) [inline content/post/054-go-ipfs-0.4.22.md]:
```suggestion
The IPFS network has scaled to the point where small changes can have a wide-reaching impact on the entire network. To keep this situation from escalating, we've put a hold on releasing new features until we can improve our [release process](https://github.com/ipfs/go-ipfs/pull/6482) (which we will be trialing in this release) and [testing procedures](https://github.com/ipfs/go-ipfs/issues/6483).
```

File: content/post/054-go-ipfs-0.4.22.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+This release includes fixes for the following regressions:
+
+1. A major bitswap throughput regression introduced in 0.4.21 ([ipfs/go-ipfs#6442](https://github.com/ipfs/go-ipfs/issues/6442)).
+2. High bitswap CPU usage when connected to many (e.g., 10,000) peers. See [ipfs/go-bitswap#154](https://github.com/ipfs/go-bitswap/issues/154).
+2. The local network discovery service sometimes initializing before the networking module, causing it to announce the wrong addresses and sometimes complain about not being able to determine the IP address) ([ipfs/go-ipfs#6415](https://github.com/ipfs/go-ipfs/pull/6415)).
```
Comment (momack2) [inline content/post/054-go-ipfs-0.4.22.md]:
```suggestion
2. The local network discovery service sometimes initializing before the networking module, causing it to announce the wrong addresses and sometimes complain about not being able to determine the IP address ([ipfs/go-ipfs#6415](https://github.com/ipfs/go-ipfs/pull/6415)).
```

File: content/post/054-go-ipfs-0.4.22.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+1. Pins not being persisted after `ipfs block add --pin` ([ipfs/go-ipfs#6441](https://github.com/ipfs/go-ipfs/pull/6441)).
+2. Concurrent map access on GC due to the pinner ([ipfs/go-ipfs#6419](https://github.com/ipfs/go-ipfs/pull/6419)).
+3. Potential pin-set corruption given a concurrent `ipfs repo gc` and `ipfs pin rm` ([ipfs/go-ipfs#6444](https://github.com/ipfs/go-ipfs/pull/6444)).
+4. Build failure due to a deleted git tag in one of our dependencies ([ipfs/go-ds-badger#64](https://github.com/ipfs/go-ds-badger/pull/65)).
+
+
+Thanks to:
+
+* @hannahhoward for fixing both bitswap issues.
```
Comment (alanshaw) [inline content/post/054-go-ipfs-0.4.22.md]: Outside of github, these are just words ;)

```suggestion
* [@hannahhoward](https://github.com/hannahhoward) for fixing both bitswap issues.
```

File: content/post/054-go-ipfs-0.4.22.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+* @sanderpick for catching and fixing the local discovery bug.
```
Comment (alanshaw) [inline content/post/054-go-ipfs-0.4.22.md]:
```suggestion
* [@sanderpick](https://github.com/sanderpick) for catching and fixing the local discovery bug.
```

File: content/post/054-go-ipfs-0.4.22.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+* @campoy for fixing the build issue.
```
Comment (alanshaw) [inline content/post/054-go-ipfs-0.4.22.md]:
```suggestion
* [@campoy](https://github.com/campoy) for fixing the build issue.
```

File: content/post/054-go-ipfs-0.4.22.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
+---
+date: 2019-08-06
+url: 054-go-ipfs-0.4.22
+title: go-ipfs 0.4.22 released
+author: Steven Allen
+---
+
+We're releasing a PATCH release of go-ipfs based on 0.4.21 containing some critical fixes.
```
Comment (alanshaw) [inline content/post/054-go-ipfs-0.4.22.md]:
```suggestion
**We've released a PATCH version of go-ipfs based on 0.4.21 containing some critical fixes.**
```

File: content/post/054-go-ipfs-0.4.22.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+The IPFS network has scaled to the point where small changes can have a wide-reaching impact on the entire network. To keep this situation from escalating, we've put a hold on releasing new features until we can improve our [release process](https://github.com/ipfs/go-ipfs/pull/6482) (which we will be trialing in this release) and [testing procedures](https://github.com/ipfs/go-ipfs/issues/6483).
```
Comment (alanshaw) [inline content/post/054-go-ipfs-0.4.22.md]: The blog post will be published after the release so makes sense to refer to the trial in the past tense?

```suggestion
The IPFS network has scaled to the point where small changes can have a wide-reaching impact on the entire network. To keep this situation from escalating, we've put a hold on releasing new features until we can improve our [release process](https://github.com/ipfs/go-ipfs/pull/6482) (which we've trialed in this release) and [testing procedures](https://github.com/ipfs/go-ipfs/issues/6483).
```

File: content/post/054-go-ipfs-0.4.22.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+This release includes fixes for the following regressions:
+
+1. A major bitswap throughput regression introduced in 0.4.21 ([ipfs/go-ipfs#6442](https://github.com/ipfs/go-ipfs/issues/6442)).
+2. High bitswap CPU usage when connected to many (e.g., 10,000) peers. See [ipfs/go-bitswap#154](https://github.com/ipfs/go-bitswap/issues/154).
```
Comment (alanshaw) [inline content/post/054-go-ipfs-0.4.22.md]:
```suggestion
2. High bitswap CPU usage when connected to many (e.g. 10,000) peers. See [ipfs/go-bitswap#154](https://github.com/ipfs/go-bitswap/issues/154).
```

File: content/post/054-go-ipfs-0.4.22.md [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+2. The local network discovery service sometimes initializing before the networking module, causing it to announce the wrong addresses and sometimes complain about not being able to determine the IP address ([ipfs/go-ipfs#6415](https://github.com/ipfs/go-ipfs/pull/6415)).
```
Comment (alanshaw) [inline content/post/054-go-ipfs-0.4.22.md]:
```suggestion
2. The local network discovery service sometimes initializes before the networking module, causing it to announce the wrong addresses and sometimes complain about not being able to determine the IP address ([ipfs/go-ipfs#6415](https://github.com/ipfs/go-ipfs/pull/6415)).
```

This reads ok as it is assuming you've read "This release includes fixes for the following regressions:" beforehand (you should have, but you might be skimming), if not then "The local network discovery service sometimes initializing" sounds incorrect. This way it makes sense both ways.

File: content/post/054-go-ipfs-0.4.22.md (hunk 6 above)
Comment (alanshaw) [inline content/post/054-go-ipfs-0.4.22.md]:
```suggestion
* @campoy for fixing the build issue.
* @stebalien for being awesome.
```

File: content/post/054-go-ipfs-0.4.22.md [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+
+# ❤️ Contributors
+
+Would you like to contribute to the IPFS project and don't know how? Well, there are a few places you can get started:
+
+- Check the issues with the `help wanted` label in the [go-ipfs repo](https://github.com/ipfs/go-ipfs/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22)
+- Join an IPFS All Hands, introduce yourself and let us know where you would like to contribute - https://github.com/ipfs/team-mgmt/#weekly-ipfs-all-hands
+- Hack with IPFS and show us what you made! The All Hands call is also the perfect venue for demos, join in and show us what you built
+- Join the discussion at [discuss.ipfs.io](https://discuss.ipfs.io/) and help users finding their answers.
+- Join the [Go Core Dev Team Weekly Sync](https://github.com/ipfs/team-mgmt/issues/650) and be part of the Sprint action!
```
Comment (alanshaw) [inline content/post/054-go-ipfs-0.4.22.md]:
```suggestion
- Join the [🚀 IPFS Core Implementations Weekly Sync 🛰](https://github.com/ipfs/team-mgmt/issues/992) and be part of the action!
```

**Pull request merged**